python_modules/dagster/dagster/core/execution/execute_in_process.py

```diff
@@ -85,3 +87,21 @@ def core_execute_in_process(
             return InProcessGraphResult(node, handle, event_list_for_top_lvl_node, recorder)
 
     check.failed(f"Unexpected node type {node}")
+
+
+def wrap_resources_for_execution(
+    resources: Optional[Dict[str, Any]] = None
+) -> Dict[str, ResourceDefinition]:
+    resources = check.opt_dict_param(resources, "resources", key_type=str)
+    resource_defs = {}
+    # Wrap instantiated resource values in a resource definition.
+    # If an instantiated IO manager is provided, wrap it in an IO manager definition.
+    for resource_key, resource in resources.items():
+        if isinstance(resource, ResourceDefinition):
+            resource_defs[resource_key] = resource
+        elif isinstance(resource, IOManager):
+            resource_defs[resource_key] = IOManagerDefinition.hardcoded_io_manager(resource)
+        else:
+            resource_defs[resource_key] = ResourceDefinition.hardcoded_resource(resource)
+
+    return resource_defs
```

Comment:
Nice. Could you also use this in `build_resources` (might be one other place where this consolidation could be used that I'm forgetting).